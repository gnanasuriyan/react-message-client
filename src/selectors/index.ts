import { createSelector } from '@reduxjs/toolkit';

import { RootState } from '~/types';

const selectSelf = (state: RootState) => state;

export const selectApp = createSelector(selectSelf, state => state.app);

export const selectMessage = createSelector(selectSelf, state => state.message);

export const selectUser = createSelector(selectSelf, state => state.user);

export const selectSignup = createSelector(selectSelf, state => state.signup);
