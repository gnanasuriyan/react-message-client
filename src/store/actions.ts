export { alertHide, alertShow } from './slices/alerts';
export { setAppOptions } from './slices/app';
export { getMessageList, getMessageListFailure, getMessageListSuccess, postMessage, postMessageSuccess, postMessageFailure } from './slices/message';
export { login, loginSuccess, logOut, logOutSuccess } from './slices/user';
export { signup, signupSuccess, signupFailure } from './slices/signup';
