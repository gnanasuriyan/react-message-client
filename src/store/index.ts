import { configureStore, Tuple } from '@reduxjs/toolkit';
import {
  FLUSH,
  PAUSE,
  PERSIST,
  persistCombineReducers,
  persistStore,
  PURGE,
  REGISTER,
  REHYDRATE,
} from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import storage from 'redux-persist/lib/storage';

import rootSaga from '~/sagas';

import { RootState } from '~/types';

import dynamicMiddlewares from './dynamic-middlewares';
import middlewares, { sagaMiddleware } from './middlewares';
import alerts, { alertsState } from './slices/alerts';
import app, { appState } from './slices/app';
import message, { messageState } from './slices/message';
import user, { userState } from './slices/user';
import signup, {signupState} from './slices/signup';

const getDefaultMiddlewareOptions = {
  serializableCheck: {
    ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
  },
  // thunk: false,
};

export const initialState = {
  alerts: alertsState,
  app: appState,
  message: messageState,
  user: userState,
  signup: signupState,
};

export const reducers = {
  alerts,
  app,
  message,
  user,
  signup
};

const rootReducer = persistCombineReducers<RootState>(
  {
    key: 'rrsb',
    stateReconciler: autoMergeLevel2,
    storage,
    blacklist: ['alerts'],
    timeout: 0,
  },
  reducers,
);

export const configStore = (preloadedState: any = {}) => {
  const enhancedStore = configureStore({
    reducer: rootReducer,
    preloadedState,
    middleware: getDefaultMiddleware => {
      return new Tuple(
        ...getDefaultMiddleware(getDefaultMiddlewareOptions),
        ...middlewares,
        dynamicMiddlewares,
      );
    },
  });

  sagaMiddleware.run(rootSaga);

  return enhancedStore;
};

export const store = configStore();
export const persistor = persistStore(store);
