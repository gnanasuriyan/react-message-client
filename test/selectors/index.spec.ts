import { selectApp, selectMessage, selectUser } from '~/selectors';
import { configStore } from '~/store';

describe('selectors', () => {
  const store = configStore();

  describe('selectApp', () => {
    it('should return the app', () => {
      expect(selectApp(store.getState())).toMatchSnapshot();
    });
  });

  describe('selectMessage', () => {
    it('should return the github', () => {
      expect(selectMessage(store.getState())).toMatchSnapshot();
    });
  });

  describe('selectUser', () => {
    it('should return the user', () => {
      expect(selectUser(store.getState())).toMatchSnapshot();
    });
  });
});
