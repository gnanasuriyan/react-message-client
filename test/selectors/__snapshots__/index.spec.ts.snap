// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`selectors > selectApp > should return the app 1`] = `
{
  "query": "react",
}
`;

exports[`selectors > selectMessage > should return the github 1`] = `
{
  "loadStatus": "idle",
  "messages": [],
  "postStatus": "idle",
}
`;

exports[`selectors > selectUser > should return the user 1`] = `
{
  "isAuthenticated": false,
  "status": "idle",
  "user": {
    "id": 0,
    "username": "",
  },
}
`;
